client/app/competitor/QuestionNavbar.tsx

```diff
@@ -0,0 +1,133 @@
+    Select,
+    SelectContent,
+    SelectGroup,
+    SelectItem,
+    SelectLabel,
+    SelectTrigger,
+    SelectValue,
+} from '@/components/ui/select';
+
+type Question = {
+    question: string;
+    description: string;
+    input: string;
+    output: string;
+    failedOutput: string;
+    expectedOutput: string;
+    status: string;
+};
+
+export default function QuestionNavbar({
+    setCurrentQuestion,
+}: {
+    setCurrentQuestion: (question: Question) => void;
+}) {
+    const [questions] = useState([
+        {
+            question: 'Sort an Array of Integers',
+            description: 'Sort an array of integers in ascending order and return it.',
+            input: '2 11 15 0',
+            output: '0 2 11 15',
+            failedOutput: '2 0 11 15',
+            expectedOutput: '0 2 11 15',
+            status: 'complete',
+        },
+        {
+            question: 'Sort an Array of Characters Alphabetically',
+            description:
+                'Sort an array of characters alphabetically and return them as a single string.',
+            input: 'h e l o',
+            output: 'e h l o',
+            failedOutput: 'h o e l',
+            expectedOutput: 'e h l o',
+            status: 'in-progress',
+        },
+        {
+            question: 'Hexadecimal in Reverse Order',
+            description:
+                'Convert characters to hexadecimal values and return them in reverse order.',
+            input: 'A B C D',
+            output: '13 12 11 10',
+            failedOutput: '10 11 12 13',
+            expectedOutput: '13 12 11 10',
+            status: 'failed',
+        },
+        {
+            question: 'Some of Digits',
+            description:
+                'Write a function that takes a positive integer as input and returns the sum of its digits.',
+            input: '1 2 3 4',
+            output: '10',
+            failedOutput: '5',
+            expectedOutput: '10',
+            status: '',
+        },
+    ]);
+    const [selectedQuestion, setSelectedQuestion] = useState<Question>(questions[0]);
+
+    const getStatusColor = (status: string) => {
+        switch (status) {
+            case 'complete':
+                return 'border-green-500 hover:border-green-300';
+            case 'in-progress':
+                return 'border-blue-500 hover:border-blue-300';
+            case 'failed':
+                return 'border-red-500 hover:border-red-300';
+            default:
+                return 'border-gray-500 hover:border-gray-300';
+        }
+    };
+
+    const getActiveColor = (status: string) => {
+        switch (status) {
+            case 'complete':
+                return 'bg-green-300 dark:bg-green-700';
+            case 'in-progress':
+                return 'bg-blue-300 dark:bg-blue-700';
+            case 'failed':
+                return 'bg-red-300 dark:bg-red-700';
+            default:
+                return 'bg-gray-300 dark:bg-gray-700';
+        }
+    };
```

Comment:
Two things:
- These should use the variables that we define in the tailwind config
- This should use a lookup table instead of a switch/case.  Look at how it is done in the leaderboard.